http/src/main/scala/Main.scala

```diff
@@ -34,33 +40,35 @@ object Main {
   private val log = Logger[this.type]
 
   def main(args: Array[String]): Unit = {
+    val singleThreadPool = ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor())
+
     val file = new File(args.headOption.getOrElse("/opt/application/conf/nelson.cfg"))
 
-    def readConfig =
-      (knobs.loadImmutable(Required(ClassPathResource("nelson/defaults.cfg")) :: Nil) |@|
-        knobs.loadImmutable(Optional(FileResource(file)) :: Nil))((a,b) =>
-        Config.readConfig(a ++ b, Http, Hikari.build _)
-      )
+    def readConfig = for {
+      defaults  <- knobs.loadImmutable[IO](Required(ClassPathResource("nelson/defaults.cfg")) :: Nil)
+      overrides <- knobs.loadImmutable[IO](Optional(FileResource(file)(singleThreadPool)) :: Nil)
```

Comment:
Needs a look - use of `singleThreadPool`

Reply:
I think I was thinking it would be good to name the thread after what it's doing, instead of an opaque thread name.